Content.Client/Pinpointer/PinpointerSystem.cs

```diff
@@ -1,78 +1,41 @@
-        {
-            base.FrameUpdate(frameTime);
-
-            // we want to show pinpointers arrow direction relative
-            // to players eye rotation (like it was in SS13)
-
-            // because eye can change it rotation anytime
-            // we need to update this arrow in a update loop
-            foreach (var pinpointer in EntityQuery<PinpointerComponent>())
-            {
-                UpdateAppearance(pinpointer.Owner, pinpointer);
-                UpdateEyeDir(pinpointer.Owner, pinpointer);
-            }
-        }
-
-        private void HandleCompState(EntityUid uid, PinpointerComponent pinpointer, ref ComponentHandleState args)
-        {
-            if (args.Current is not PinpointerComponentState state)
-                return;
+namespace Content.Client.Pinpointer;
 
-            pinpointer.IsActive = state.IsActive;
-            pinpointer.ArrowAngle = state.ArrowAngle;
-            pinpointer.DistanceToTarget = state.DistanceToTarget;
-        }
-
-        private void UpdateAppearance(EntityUid uid, PinpointerComponent? pinpointer = null,
-            AppearanceComponent? appearance = null)
-        {
-            if (!Resolve(uid, ref pinpointer, ref appearance))
-                return;
-
-            _appearance.SetData(uid, PinpointerVisuals.IsActive, pinpointer.IsActive, appearance);
-            _appearance.SetData(uid, PinpointerVisuals.TargetDistance, pinpointer.DistanceToTarget, appearance);
-        }
+public sealed class PinpointerSystem : SharedPinpointerSystem
+{
+    [Dependency] private readonly IEyeManager _eyeManager = default!;
 
-        private void UpdateArrowAngle(EntityUid uid, Angle angle, PinpointerComponent? pinpointer = null,
-            AppearanceComponent? appearance = null)
-        {
-            if (!Resolve(uid, ref pinpointer, ref appearance))
-                return;
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
 
-            _appearance.SetData(uid, PinpointerVisuals.ArrowAngle, angle, appearance);
-        }
+        // we want to show pinpointers arrow direction relative
+        // to players eye rotation (like it was in SS13)
 
-        /// <summary>
-        ///     Transform pinpointer arrow from world space to eye space
-        ///     And send it to the appearance component
-        /// </summary>
-        private void UpdateEyeDir(EntityUid uid, PinpointerComponent? pinpointer = null)
+        // because eye can change it rotation anytime
+        // we need to update this arrow in a update loop
+        var query = EntityQueryEnumerator<PinpointerComponent, SpriteComponent>();
+        while (query.MoveNext(out var _, out var pinpointer, out var sprite))
         {
-            if (!Resolve(uid, ref pinpointer) || !pinpointer.HasTarget)
+            if (!pinpointer.HasTarget)
                 return;
-
             var eye = _eyeManager.CurrentEye;
             var angle = pinpointer.ArrowAngle + eye.Rotation;
```

Comment:
Won't this break with multi-viewport.

Reply:
e.g. client can have cameras open in a viewport which uses a different eye, so those pinpointers will look incorrect.

Reply:
well, i just used the old code, maybe fix it later